components/heap/heap_trace_standalone.c

```diff
@@ -161,38 +198,31 @@ esp_err_t heap_trace_get(size_t index, heap_trace_record_t *r_out)
     } else {
 
         // Perf: speed up sequential access
-        if (r_get && r_get_idx == index - 1) {
+        if (rget && rget_idx == index - 1) {
 
-            r_get = TAILQ_NEXT(r_get, tailq);
-            r_get_idx = index;
+            rget = TAILQ_NEXT(rget, tailq);
+            rget_idx = index;
 
         } else {
 
             // Iterate through through the linked list
 
-            r_get = TAILQ_FIRST(&records.list);
+            rget = TAILQ_FIRST(&records.list);
 
             for (int i = 0; i < index; i++) {
 
-                if (r_get == NULL) {
+                if (rget == NULL) {
                     break;
                 }
 
-                r_get = TAILQ_NEXT(r_get, tailq);
-                r_get_idx = i + 1;
+                rget = TAILQ_NEXT(rget, tailq);
+                rget_idx = i + 1;
             }
         }
 
-        // copy to destination
-        if (r_get) {
-            memcpy(r_out, r_get, sizeof(heap_trace_record_t));
-        } else {
-            // this should not happen since we already
-            // checked that index < records.count,
-            // but could be indicative of memory corruption
-            result = ESP_ERR_INVALID_STATE;
-            memset(r_out, 0, sizeof(heap_trace_record_t));
-        }
+        // We already checked that index < records.count,
+        // This could be indicative of memory corruption.
+        assert(rget != NULL);
```

Comment:
Is it normal that the copy of `r_get`  to `r_out` was removed here ?

Reply:
Oh good spot. That is a bug.

I only intended to remove the error code, because I changed my mind. I think asserting on memory corruption is more appropriate.